packages/account/src/providers/transaction-request/utils.ts

```diff
@@ -51,3 +61,13 @@ export const isTransactionTypeCreate = (
 export const isTransactionTypeBlob = (
   request: TransactionRequestLike
 ): request is BlobTransactionRequest => request.type === TransactionType.Blob;
+
+/** @hidden */
+export const isTransactionTypeUpgrade = (
+  request: TransactionRequestLike
+): request is UpgradeTransactionRequest => request.type === TransactionType.Upgrade;
+
+/** @hidden */
+export const isTransactionTypeUpload = (
+  request: TransactionRequestLike
+): request is UpgradeTransactionRequest => request.type === TransactionType.Upload;
```

Comment:
```suggestion
): request is UploadTransactionRequest => request.type === TransactionType.Upload;
```

You should also add some tests for this in the [utils.test.ts](https://github.com/FuelLabs/fuels-ts/blob/master/packages/account/src/providers/transaction-request/utils.test.ts#L13-L47) file

Reply:
We'll certainly need some integration tests via `launchTestNode` as well. ([example](https://github.com/FuelLabs/fuels-ts/blob/682d62eb7c45cfb04816378b17453a7e5e25fd2b/packages/fuel-gauge/src/transaction-response.test.ts#L105))

You can find several other examples in the repo; search for `await launchTestNode`.

Also, check the docs:
 - https://docs.fuel.network/docs/fuels-ts/testing/launching-a-test-node/

> cc @Torres-ssf - in case you have any tips or special requests in this regard.

Reply:
We will definitely benefit from integration tests for these 2 types. 

@luisburigo, you can use `launchTestNode` to achieve this, as it supports starting fuel-core with the [necessary flags](https://github.com/FuelLabs/fuels-ts/blob/309b8d5d40ffc1cda3cdd5b44406ed0b2e071a5a/packages/fuel-gauge/src/funding-transaction.test.ts#L437) and setting [chain configuration values](https://github.com/FuelLabs/fuels-ts/blob/309b8d5d40ffc1cda3cdd5b44406ed0b2e071a5a/packages/account/src/providers/provider.test.ts#L749). Let us know if you have any questions.

Reply:
Perfect, guys! thanks for the links
I will implement the tests. what would be the best location to implement them?
in [transaction-request.test.ts](https://github.com/FuelLabs/fuels-ts/blob/cf25280461e21b0856e87d118ff87bf80df30ba6/packages/account/src/providers/transaction-request/transaction-request.test.ts) or other location?

Reply:
@luisburigo It can be at `packages/fuel-gauge/src/transaction.test.ts`